.github/workflows/pytest-coverage.yaml

```diff
@@ -15,7 +15,24 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.10", "3.11", "3.12", "3.13"]
+        version:
+            # Test the lower bound of pandas/numpy support for Python 3.9,
+            # and test the upper bound for all other Python versions
+          - python: "3.9"
+            pandas: "1.4.*"
+            numpy: "1.23.*"
+          - python: "3.10"
+            pandas: "2.2.*"
+            numpy: "2.1.*"
+          - python: "3.11"
+            pandas: "2.2.*"
+            numpy: "2.1.*"
+          - python: "3.12"
+            pandas: "2.2.*"
+            numpy: "2.1.*"
+          - python: "3.13"
+            pandas: "2.2.*"
+            numpy: "2.1.*"
```

Comment:
This matrix feels a little bit clunky to me, given that we only care about specifying the pandas/numpy versions for Python 3.9, but I couldn't think of a clearer way of specifying it. Any ideas?